test/retex_test.exs

```diff
@@ -169,6 +169,64 @@ defmodule RetexTest do
              ] = agenda
     end
 
+    test "return only the bindings of the fully activated rule" do
+      given = [
+        has_attribute(:Account, :status, :==, "$a"),
+        has_attribute(:Account, :premium, :==, "$b"),
+        has_attribute(:Account, :age, :==, "$a"),
+        has_attribute(:Account, :age, :>, 21)
+      ]
+
+      action = [
+        {:Account, :activated_1, "$a"}
+      ]
+
+      rule = create_rule(lhs: given, rhs: action)
+
+      given_2 = [
+        has_attribute(:Account, :status, :==, "$a"),
+        has_attribute(:Account, :status, :!==, :blue),
```

Comment:
This should impede the binding `%{"$a" => :blue}` from reaching the activation node but it does not work at the moment